src/components/post-card/PostCard.js

```diff
@@ -4,61 +4,90 @@ import { Comment } from '../comment/Comment';
 import DefaultImg from '../../assets/default-empty-img.png';
 import './PostCard.scss';
 
-// todo: переписать эту компоненту, чтоб она стала классовой
+// donetodo: переписать эту компоненту, чтоб она стала классовой
 //  в ней должно появиться свойство state
 //  в котором будет флажок showComments
 //  и будет кнопка, лейба которой будет либо "show comments" либо "hide comments"
 
-export function PostCard(props) {
-  const { post, hasImage, comments = [], author } = props;
+export class PostCard extends React.Component {
+  constructor (props) {
+    super(props);
+    const {post, hasImage, comments = [], author} = this.props;
+    this.post = post;
```

Comment:
This is not a good approach to save props as inner properties
The idea is that based on the change of the props there should rerender happen. a constructor is called only once. And when you write all the props to `this` and then use properties from `this` in render  - there wouldn't be rerender when needed
For example, if you already saw the lecture 4, there is only one PostCard used for displaying different posts. If to use this mechanism of saving props inside this - it would work

Reply:
Добрый день. Переделал PostCard.js